Node.js/GettingStarted/azure_batch_client.js

```diff
@@ -0,0 +1,196 @@
+                     console.log("Pool created successfully now submitting job");
+                     createJob();
+                  }
+                  //// Create the Job 
+            }
+        }
+        else
+        {
+            if(error.statusCode==404)
+            {
+                //console.log("Pool not found yet returned 404...");    
+                           
+            }
+            else
+            {
+                poolCreationError = true;
+            }
+        }
+        });
+  
+  numRetries++;
+ 
+}
+
+console.log("Pool Status : " + isPoolStateActive);
+
+
+
+
+
+
+function createJob()
+{
+    console.log("Inside create job...");
+    var sh_url = "Shell script SAS URI";
+    var job_prep_task_config = {id:"installprereq",commandLine:"sudo sh startup_prereq.sh > startup.log",resourceFiles:[{'blobSource':sh_url,'filePath':'startup_prereq.sh'}],waitForSuccess:true,runElevated:true}
+    // Setting up Batch pool configuration
+    var poolConfigJob = {poolId:poolid}
+    
+    
+    var job_config = {id:jobId,displayName:"process csv files",jobPreparationTask:job_prep_task_config,poolInfo:poolConfigJob}
+    
+     // Adding Azure batch job to the pool
+     var job = batch_client.job.add(job_config,function(error,result){
+        if(error !=null)
+        {
+            console.log(error);
+        }        
+        
+    });       
+    var maxRetryJob = 100;
+    var isJobCreated = false
+    var isJobCreatedError = false
+    var numRetriesJob = 0;
+    var taskSubmitted = false;
+    while(numRetriesJob < maxRetryJob && isJobCreated == false && isJobCreatedError == false)
+    {
+        var jobStatus = batch_client.job.get(jobId,function(error,result,request,response)
+        {
+            if(error == null)
+            {
+               // console.log(result);
+               // console.log("Result state" + result.state);
+                if(result.state == 'active')
+                {
+                    
+                      isJobCreated = true;  
+                      if (taskSubmitted==false)
+                      {
+                          console.log("Job found...");
+                          console.log("Creating tasks...");
+                          createTasks();
+                          taskSubmitted = true;
+                      }
+                }
+            }
+            else
+            {
+                if(error.statusCode == 404)
+                {
+                    //console.log("Job not created yet, received 404...");
+                }
+                else
+                {
+                    console.log("Error occured while job creation" + error.response);
+                    isJobCreatedError = true;
+                }
+            }
+        }); 
+        numRetriesJob++;
+   }
+    
+
+}
+
+function createTasks()
+{
+    var container_list = ["con1","con2","con3","con4"]
+    var scriptURI = "Python scrip SAS URI";
```

Comment:
'script' - also consider pulling out to a configuration variable at the top of the file